include/openthread/thread.h

```diff
@@ -781,6 +781,37 @@ void otThreadRegisterParentResponseCallback(otInstance *                   aInst
                                             otThreadParentResponseCallback aCallback,
                                             void *                         aContext);
 
+/**
+ * This structure represents the Thread Discovery Request data.
+ *
+ */
+typedef struct otThreadDiscoveryRequestInfo
+{
+    const otExtAddress *mExtAddress;   ///< IEEE 802.15.4 Extended Address of the requester
+    uint8_t             mVersion : 4;  ///< Thread version.
+    bool                mIsJoiner : 1; ///< Whether is from joiner.
+} otThreadDiscoveryRequestInfo;
```

Comment:
@bukepo, the implementation overall looks good.

Can you please help give a bit more detail on how we can use this (how it would help)?

One thing is that, in current code when a joiner requests a discover scan, it uses a randomly generated ext address and not the Joiner ID (was recently changed from #4920) which I guess required by spec (?). So we cannot really tell apart joiners when they issue scan from their ext-address.

Reply:
Yes, #4920 was required by the Spec.

Reply:
This is useful when https://github.com/openthread/openthread/issues/5175 is implemented.